packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol

```diff
@@ -401,6 +451,21 @@ contract TokenFacet is BaseConnextFacet {
     if (s.canonicalToAdopted[_key] != _adoptedAssetId || s.canonicalToRepresentation[_key] != _representation)
       revert TokenFacet__removeAssetId_invalidParams();
 
+    bool onCanonical = s.domain == _canonical.domain;
+    if (onCanonical) {
+      // Sanity check: no value custodied if on canonical domain
+      address canonicalAsset = TypeCasts.bytes32ToAddress(_canonical.id);
+      // Check custodied amount for the given canonical asset address.
+      if (s.custodied[canonicalAsset] > 0) {
```

Comment:
if the `cap` is 0 then `custodied[]` isn't tracked  (so maybe better to use the balanceOf)